Migrate to  node@4.4.2, eslint@2.7.0, babel-eslint@6.0.2 and eslint-p…

liubko opened the pull request:
…lugin-react@4.3.0

@playpauseandstop please take a look on the list of rules.
1. I'm not sure which rules should be in `base`, and which should be moved to `alternate`.
2. do you think we should add any of this rules? ```"react/jsx-no-bind", "react/forbid-prop-types", "react/no-string-refs"```
3. what is `node_modules/eslint-config-ezhome` for?

liubko commented:
Updated PR:
- use `eslint-plugin-import-order@2.1.2` plugin _(https://github.com/jfmengels/eslint-plugin-import-order)_

playpauseandstop commented:
@liubko 

Thanks. Let discuss this tomorrow as well.

liubko commented:
@playpauseandstop ^

playpauseandstop commented on package.json:
```diff
@@ -1,7 +1,6 @@
   "name": "eslint-config-ezhome",
-  "version": "0.6.1",
-
+  "version": "1.0.0",
```
Please list this as `1.0.0-beta.1` version. I'd like to try it, but not sure that all necessary rules integrated yet.

playpauseandstop commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+1.0.0 (2016-04-21)
```
> 1.0.0 (Not Released Yet)

playpauseandstop commented:
LGTM

playpauseandstop commented:
BTW, please check if it possible to update for even further versions. Thanks

liubko commented:
> BTW, please check if it possible to update for even further versions. Thanks

Updated eslint-plugin-react to 5.1.1
- added new rule `eact/jsx-first-prop-new-line`
- added new rule `eact/jsx-no-target-blank`

playpauseandstop commented on README.md:
```diff
@@ -22,11 +22,11 @@ This package provides Ezhome's `.eslintrc` as an extensible shared config.
-* `eslint@1.10.3`
-* `babel-eslint@4.1.6`
-* `eslint-plugin-react@3.11.2`
+* `eslint@2.7.0`
```
2.9.0

playpauseandstop commented on README.md:
```diff
@@ -22,11 +22,11 @@ This package provides Ezhome's `.eslintrc` as an extensible shared config.
-* `babel-eslint@4.1.6`
-* `eslint-plugin-react@3.11.2`
+* `eslint@2.7.0`
+* `babel-eslint@6.0.2`
```
6.0.4

playpauseandstop commented on README.md:
```diff
@@ -22,11 +22,11 @@ This package provides Ezhome's `.eslintrc` as an extensible shared config.
-* `eslint-plugin-react@3.11.2`
+* `eslint@2.7.0`
+* `babel-eslint@6.0.2`
+* `eslint-plugin-react@4.3.0`
```
5.1.1

playpauseandstop commented on package.json:
```diff
@@ -22,19 +21,17 @@
   },
   "eslintConfig": {
-    "extends": "ezhome/base"
+    "extends": "ezhome/base/index.js"
```
This should be reflected in README as well, if we need to extend exact modules, not packages as before

liubko replied:
@playpauseandstop not sure I understand you. this is to lint the repo itself. 
to extend from many we could do ```"extends": ["ezhome/base/index.js", "..."] or make changes to`ezhome/base/index.js` itself.

playpauseandstop commented:
Please make an `v1.0.0-beta.1` tag and push it to GitHub, to create new version on NPM. Thanks

liubko commented:
> Please make an v1.0.0-beta.1 tag and push it to GitHub, to create new version on NPM. Thanks

done
